src/components/answer/getResult.js

```diff
@@ -0,0 +1,51 @@
+import { MissionUtils } from '@woowacourse/mission-utils';
+import { RESULT_MESSAGE } from '../../constants';
+
+const getStrike = (computerNumbers, userNumbers) => {
+  userNumbers = userNumbers.split('').map(Number);
+  return computerNumbers.reduce((acc, currentValue, index) => {
+    if (currentValue === userNumbers[index]) {
+      return acc + 1;
+    } else return acc;
+  }, 0);
+};
+
+const getBall = (computerNumbers, userNumbers) => {
+  userNumbers = userNumbers.split('').map(Number);
+  return computerNumbers.reduce((acc, currentValue, index) => {
+    if (
+      currentValue !== userNumbers[index] &&
+      computerNumbers.includes(userNumbers[index])
+    ) {
+      return acc + 1;
+    } else return acc;
+  }, 0);
+};
+
+const printResult = (ball, strike) => {
+  if (strike === 3) {
+    MissionUtils.Console.print(RESULT_MESSAGE.ANSWER);
+    return true;
+  }
+  if (!strike && !ball) {
+    MissionUtils.Console.print(RESULT_MESSAGE.NOTHING);
+  }
+  if (!strike && ball) {
+    MissionUtils.Console.print(`${ball}볼`);
+  }
+  if (strike && !ball) {
+    MissionUtils.Console.print(`${strike}스트라이크`);
+  }
+  if (strike && ball) {
+    MissionUtils.Console.print(`${ball}볼 ${strike}스트라이크`);
+  }
```

Comment:
이렇게 if 구문만 사용하게 되면 모든 if 문의 조건을 탐색하게 됩니다!
즉 위 코드에서 첫번째 `!strike && !ball` 이 참일지라도 그 아래 3개의 조건문을 모두 검사하게 되는거죠.

이 부분도 이렇게 바꿔볼 수 있겠네요!
```suggestion
  if (!strike && !ball) {
    MissionUtils.Console.print(RESULT_MESSAGE.NOTHING);
  } else if (!strike && ball) {
    MissionUtils.Console.print(`${ball}볼`);
  } else if (strike && !ball) {
    MissionUtils.Console.print(`${strike}스트라이크`);
  } else {
    MissionUtils.Console.print(`${ball}볼 ${strike}스트라이크`);
  }
```

Reply:
`printResult`함수 같은 경우, 이전에 비교 결과 메세지만을 return하는 함수였었는데 명칭 분리와 함수 별 역할 분리에 어려움을 겪다 Console.print에 통합하면서 어려움이 발생한 것 같습니다! 언급해주신대로, 객체와 함수가 하나의 기능을 명확히 설명할 수 있도록 수정해보도록 하겠습니다...! 짚어주셔서 감사해요~! 

// 기존에 작성한 코드
![image](https://github.com/woowacourse-precourse/javascript-baseball-6/assets/51940808/6940aff9-2321-4c8b-bed2-6b9032ab4358)
[참고한 no-else-return airbnb convention](https://github.com/airbnb/javascript#blocks--no-else-return)